File: pset1/greedy.c
/*
-- Greedy --
K.Yap August 2016
Finds the least number of coins 'greedy algorithm' that can be given for a user defined amount of money
*/

#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float amnt;
    do
    {
        printf("How much $$ are you owed?");
        amnt = GetFloat();
    }
    while (amnt < 0);
    
    int iamnt = round(amnt*100);
    int count = 0;
    int q = 25;
    int d = 10;
    int n = 5;
    int p = 1;
    
    //printf("The amount before loop is %i\n", iamnt);
    
        while (iamnt - q >= 0)
        {
            iamnt-=q;
            ++count;
        }
        while (iamnt - d >= 0)
        {
            iamnt-=d;
            ++count;
        }        
        while (iamnt - n >= 0)
        {
            iamnt-=n;
            ++count;
        }        
        while (iamnt - p >= 0)
        {
            iamnt-=p;
            ++count;
        }

    //printf("The amount is %i\n", iamnt);
    printf("%i\n", count);
 
}

File: pset6/greedy.py
#-- Greedy in Python --
#K.Yap March 2017
#Finds the least number of coins 'greedy algorithm' that can be given for a user defined amount of money
#
#american coin values for reference:
#q = 25
#d = 10
#n = 5
#p = 1

#ask for input and check it is a positive, non-zero integer
while True:
    try:
        amnt = float(input("How much $$$ are you owed?"))
    except ValueError:
        print("Please enter a dollar amount")
        continue
    if (amnt < 0):
        print("Please enter an amount greater than 0")
        continue
    else:
        break
    
iamnt = 100*amnt
count = 0
coins = [25 , 10, 5, 1]
c = 0   #to traverse the coins list

while (iamnt - coins[c] >= 0):
    iamnt-=coins[c]
    count+=1
    
    if (iamnt - coins[c] >= 0):
        continue
    elif (iamnt == 0):
        break
    else:
        c+=1
        continue

print(count)
